contessa/executor.py

```diff
@@ -67,14 +67,26 @@ def compose_where_time_filter(self, rule):
         only data that were updated/created/confirmed in last 30 days.
         :return: str, WHERE `time_filter` filter statement
         """
-        time_filter_column = rule.time_filter
-        if time_filter_column:
-            past = (datetime.now() - timedelta(days=30)).strftime(
-                "%Y-%m-%d %H:%M:%S UTC"
-            )
-            return f"""{time_filter_column} >= '{past}'::timestamp"""
-        else:
-            return ""
+        days = 30
+        filters = []
+        result = []
+        if rule.time_filter:
+            if isinstance(rule.time_filter, str):
+                filters.append({"column": rule.time_filter, "days": days})
+            elif isinstance(rule.time_filter, list):
+                for each in rule.time_filter:
+                    filters.append(
+                        {"column": each.get("column"), "days": each.get("days", days)}
+                    )
+
+            for each in filters:
+                past = (
+                    self.context.get("task_ts", datetime.now())
```

Comment:
I think it safer to put `task_ts` default at one place. Maybe this is the first one :D, but i think there was some default setting before. 

I would make context before executor or do it in the `get_context` method, or I would consider making it a property. But probably the get_context method is good place to start.

Reply:
Default for _task_ts_ is set in the *runner.get_context* method. Main reason why I put it also here, was my concern that _runner_ is not always only entry point. If it is, it can be removed from this place.

Reply:
ok i got that. we cannot rely on that i think. pls, do one of:

1. let's extract construction of default context and use that in any place needed. 
2. construct default context only in Executor and nowhere else.

I prefer 1., but its up to you.

Reply:
As discussed, *context* will be constructed once in the _runner_ and then used everywhere else.